fluency-fluentd/src/main/java/org/komamitsu/fluency/fluentd/FluencyBuilderForFluentd.java

```diff
@@ -117,24 +117,34 @@ public void setReadTimeoutMilli(Integer readTimeoutMilli)
         this.readTimeoutMilli = readTimeoutMilli;
     }
 
+    public FluentdRecordFormatter getRecordFormatter()
+    {
+        return recordFormatter;
+    }
+
+    public void setFluentdRecordFormatter(FluentdRecordFormatter recordFormatter)
```

Comment:
The intention of accepting`FluentdRecordFormatter` not `RecordFormatter`is to avoid using a wrong formatter class?

👍

Reply:
The method name seems a bit verbose. How about just `setRecordFormatter()`? We can add the same name to other FluencyBuilders if needed.

Reply:
Setter has been renamed as suggested. The FluentdRecordFormatter has been used to avoid using a wrong formatter.

Custom formatter could be available for every builders but to be consistent, this would require a quite important rework of the builders which looks out of scope for this issue. 

By the way, the FluencyBuilderForAwsS3#build(AwsS3RecordFormatter recordFormatter) already supports custom recorders but the FluencyBuilderForTreasureData doesn't.

Ideally, we may have an abstract FluencyBuilder with some generic and sub implementations with a set of with* methods to finally build Fluency using a single build method.
```
FluencyBuilderForFluentd.create(String host, Integer port)
   .withRecordFormatter(new FluentdRecordFormatter()) if not set, default would be the same as now
   .withSender(new FluentdSender()) // if not set, default would be the same as now
   .withReadTimeoutMilli(100)
   .with* 
   .build();
```

Same for other builders. This may simplify a lot the code of builders.

Reply:
> the FluencyBuilderForAwsS3#build(AwsS3RecordFormatter recordFormatter) already supports custom recorders

I don't think it's for custom recorders. There are 3 types of AwsS3RecordFormatter corresponding to MessagePack, JSON and CSV formats and users can choose one of them by passing one of them.